src/edgeDebugAdapter.ts

```diff
@@ -60,7 +61,47 @@ export class EdgeDebugAdapter extends CoreDebugAdapter {
         return capabilities;
     }
 
-    public launch(args: ILaunchRequestArgs): Promise<void> {
+    private async _checkPortOccupiedByEDP(address: string, port: number): Promise<void> {
+        // See if EDP is using the port, if that's the case we can still launch
+        // We see if we can hit /json/version endpoint to verify if EDP is running on the port
+        const url = `http://${address}:${port}/json/version`;
+        logger.log(`Checking if EDP is running on the port by hitting ${url}`);
+
+        const jsonResponse = await coreUtils.getURL(url, { headers: { Host: 'localhost' } })
+            .catch(e => {
+                // This means /json/version is not available. So the port is being used by another process.
+                // Error out in this case.
+                logger.log(`There was an error connecting to ${url} : ${e.message}`);
+                telemetry.telemetry.reportEvent('portOccupiedByAnotherProcess', port);
+                return coreUtils.errP(localize('edge.port.occupied', 'Cannot launch Edge. The specified debugging port {0} is in use by another process.', port));
```

Comment:
Do you think the user will understand what action to take to get the debugger working based on this message?

Reply:
I added some more context :)